src/components/tl/ucp/scatter/scatter_knomial.c

```diff
@@ -174,6 +174,16 @@ ucc_status_t ucc_tl_ucp_scatter_knomial_start(ucc_coll_task_t *coll_task)
     task->scatter_kn.recv_dist = ucc_knomial_calc_recv_dist(size - p->n_extra,
                                                             vrank, p->radix,
                                                             vroot);
+    task->scatter_kn.recv_offset = 0;
+    is_zcopy = UCC_TL_UCP_TEAM_LIB(team)->cfg.scatter_kn_enable_recv_zcopy;
+    if (((is_zcopy == UCC_CONFIG_AUTO) &&
+         (args->src.info.mem_type != UCC_MEMORY_TYPE_HOST)) ||
```

Comment:
why not on host?

Reply:
this tricky, depending on CPU vendor I see different results. For AMD it's better to have extra copy and for Intel it's better to receive with correct offset. For GPUs it's always better to avoid copy